.circleci/continue_config.yml

```diff
@@ -284,14 +279,12 @@ jobs:
           # Setting it to 2 or higher will suppress the warning messages totally.
           name: Set HDF5_DISABLE_VERSION_CHECK environment variable
           command: setx /m HDF5_DISABLE_VERSION_CHECK 1
-      - unless:
+      - when:
           condition:
-            or:
-              - and:
-                - equal: [ "3.10", <<parameters.python_version>> ]
-                - equal: [true, <<pipeline.parameters.run-build-kedro-datasets>>]
-              - or:
-                - equal: [false, <<pipeline.parameters.run-build-kedro-datasets>>]
+            and:
+              - not:
+                  equal: [ "3.10", <<parameters.python_version>> ]
+              - equal: [true, <<pipeline.parameters.run-build-kedro-datasets>>]
```

Comment:
My brain always gets twisted when I read CCI config.

Reply:
@noklam TBH I feel like some of this can be just done in bash. For example, whether or not it's parallel, the step can be named "Run unit tests without spark", and then whether sequential or parallel can just be decided in the job itself. Reduce the config complexity in a way, hopefully wouldn't make errors harder to debug?

Reply:
I think this can be done with/without bash? This is a consequence that you want to run these tests locally too, if that logic is embedded with CI job parameters, you can't do it locally with the `Makefile`.